### PR TITLE
Shared DataSource instances across maha service

### PR DESCRIPTION
**Pull request by pranavbhole**

### CODE REVIEW

File: service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala [hunk 1]
```diff
@@ -146,28 +160,36 @@ class PrestoQueryExecutoryFactory extends QueryExecutoryFactory {
...
-      dataSourceFactoryConfig <- dataSourceFactoryConfigResult
-      dataSourceFactory <-  getFactory[DataSourceFactory](dataSourceFactoryClass, this.closer)
-      dataSource <- dataSourceFactory.fromJson(dataSourceFactoryConfig)
+      dataSourceName <- dataSourceNameResult
+      dataSource <- context.dataSourceMap.get(dataSourceName).successNel[Option[DataSource]].asInstanceOf[MahaServiceConfig.MahaConfigResult[Option[DataSource]]]
       jdbcConnectionFetchSizeOption <- jdbcConnectionFetchSizeOptionResult
     } yield {
       if(jdbcConnectionFetchSizeOption.isDefined) {
-        new JdbcConnection(dataSource, jdbcConnectionFetchSizeOption.get)
+        new JdbcConnection(dataSource.get, jdbcConnectionFetchSizeOption.get)
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala]: This is unsafe get on dataSource.  Check if it is defined first.

File: service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala [hunk 2]
```diff
@@ -146,28 +160,36 @@ class PrestoQueryExecutoryFactory extends QueryExecutoryFactory {
...
       } else {
-        new JdbcConnection(dataSource)
+        new JdbcConnection(dataSource.get)
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala]: Same here, unsafe get

File: service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala (hunk 1 above)
Comment (pranavbhole) [reply service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala]: datasource.get is already validated above if not defined, execution return the MahaServiceError list.

File: service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala (hunk 2 above)
Comment (pranavbhole) [reply service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala]: same here

File: service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala (hunk 1 above)
Comment (patelh) [reply service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala]: Can you add a test for this?

Comment (pranavbhole) [reply service/src/main/scala/com/yahoo/maha/service/factory/QueryExecutoryFactory.scala]: Added tests for checking unknown datasource

**Pull request merged**